tests/manage/test_create_pvc_random_storage_class.py

```diff
@@ -85,17 +83,19 @@ def create_multiple_rbd_storageclasses(count=1):
     return True
 
 
-def create_pvc(storageclass_list, count=1):
+def create_pvc(storageclass_list, pvcs, count=1):
```

Comment:
I think what I'd probably do here is make a fixture out of ``create_pvc`` that is essentially this same code and returns a list of pvc objects. This "factory function" fixture can then handle the teardown for each pvc that it creates.

This way the tests that need to create pvcs can just request the fixture, call it  with a count to create as many pvcs as they need and can depend on pytest cleaning things up with finalizers.

Usage might look something like:

```
def test_foo(create_pvcs):
    pvcs = create_pvcs(count=20)
    # perform tests on the pvcs
```

Here's some docs on fixtures as factory functions that's a good reference: https://docs.pytest.org/en/latest/fixture.html#factories-as-fixtures

Reply:
It makes sense to use functions and return them, we could easily reuse those functions for multiple cases, I am bit surprised how many tests are creating pv, pvcs, storageclass etc, when they could be just couple of reusable functions.

Reply:
I agree with @andrewschoen

Reply:
There's one problem with this function factory and is the fact that the creation of objects should happen in the setup and not part of the test.
In thi case, pvcs should be created in a setup phase and not be considered as part of the test

Reply:
@RazTamir Would something like this work for you?

```python
@pytest.fixture()
def multiple_pvcs(create_pvcs):
    return create_pvcs(count=20)

def test_foo(multiple_pvcs):
    for pvc in multiple_pvcs:
        pvc.do_whatever()
```

This keeps the creation of pvcs out of the test logic while benefitting from the customization of the factory style fixture. It just takes having another fixture which the test uses actually utilize the factory fixture to create them.

Reply:
@clacroix12 , with that usage, how can I pass to multiple_pvcs() more parameters that will depend on the test needs? for example the namespace of the PVCs, size, etc.

Reply:
Fixture `multiple_pvcs` from the example above would be specific for the particular test(s), so that if the test(s) needs more parameters, you will specify it in the fixture. This is in contrast with `create_pvcs`, which is very general.